src/MetaMorpho.sol

```diff
@@ -386,14 +386,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             totalSupplied += suppliedAssets;
         }
 
-        if (totalWithdrawn > totalSupplied) {
-            idle += totalWithdrawn - totalSupplied;
-        } else {
-            uint256 idleSupplied = totalSupplied - totalWithdrawn;
-            if (idle < idleSupplied) revert ErrorsLib.InsufficientIdle();
-
-            idle -= idleSupplied;
-        }
+        if (totalWithdrawn != totalSupplied) revert ErrorsLib.InconsistentReallocation();
```

Comment:
Indeed I suspect there will be

Reply:
I think it's safe because there's always a way to make it not reverting.

With:
```solidity
if (allocation.assets == type(uint256).max) allocation.assets = totalWithdrawn.zeroFloorSub(totalSupplied);
```

you'll supply exactly `totalWithdrawn.zeroFloorSub(totalSupplied)` and `suppliedAssets` will be equal to this value.